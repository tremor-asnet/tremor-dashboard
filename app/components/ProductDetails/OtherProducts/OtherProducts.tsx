"use client";

// Components
import {
  CustomAvatarName,
  CustomNumberFormat,
} from "@/components/Table/common";
import { DataGrid } from "@/components";

// Types
import { Product, ColumnType } from "@/types";

// Constants
import { ProgressBar } from "@tremor/react";

//Styles
import "@/styles/products.css";

interface OtherProductsProps {
  products: Product[];
  isAvailable: string;
  keyword: string;
  className?: string;
  hasPagination?: boolean;
  hasSort?: boolean;
}

const OtherProducts = ({
  products,
  isAvailable,
  keyword,
  className = "",
  hasPagination = true,
  hasSort = true,
}: OtherProductsProps) => {
  // Other Product Table Props
  const columns: ColumnType<Product>[] = [
    {
      key: "product",
      title: "Product",
      customNode: (_, { productName, image }) => (
        <CustomAvatarName avatar={image} text={productName} />
      ),
    },
    {
      key: "price",
      title: "Price",
      customNode: (_, { price }) => <CustomNumberFormat value={price} />,
    },
    {
      key: "isAvailable",
      title: "Availability",
      customNode: (_, { isAvailable }) => (
        <p className="text-xs dark:text-lighter font-semibold product-availability">
          {isAvailable ? (
            <ProgressBar value={80} color="green" className="w-32" />
          ) : (
            <ProgressBar value={60} color="orange" className="w-32" />
          )}
        </p>
      ),
    },
    {
      key: "id",
      title: "Id",
    },
  ];

  return (
    <DataGrid
      data={products}
      columns={columns}
      filterBy={isAvailable}
      keyword={keyword}
      className={className}
      hasPagination={hasPagination}
      hasSort={hasSort}
    />
  );
};

export default OtherProducts;
