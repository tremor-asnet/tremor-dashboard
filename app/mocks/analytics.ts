export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
