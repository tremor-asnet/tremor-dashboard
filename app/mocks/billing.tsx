// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
