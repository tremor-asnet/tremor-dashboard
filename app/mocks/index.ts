export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
