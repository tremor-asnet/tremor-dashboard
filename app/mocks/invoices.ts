import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
