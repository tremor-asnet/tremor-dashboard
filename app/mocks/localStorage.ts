// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
