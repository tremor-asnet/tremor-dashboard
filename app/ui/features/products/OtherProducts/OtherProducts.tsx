"use client";

// Components
import {
  DataGrid,
  CustomAvatarName,
  CustomNumberFormat,
} from "@/ui/components";
import { Text } from "@tremor/react";
import Link from "next/link";

// Types
import { Product, ColumnType } from "@/types";

// Constants
import { ROUTES } from "@/constants";

// Styles
import "@/styles/products.css";

export const OtherProducts = ({ products }: { products: Product[] }) => {
  // Other Product Table Props
  const columns: ColumnType<Product>[] = [
    {
      key: "id",
      title: "Id",
      customNode: (_, { id }) => (
        <Link
          prefetch={true}
          href={`${ROUTES.PRODUCT_LIST}/${id}`}
          className="text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] order-id hover:underline">
          &#35;{id}
        </Link>
      ),
    },
    {
      key: "product",
      title: "Product",
      customNode: (_, { productName, image }) => (
        <CustomAvatarName avatar={image} text={productName} />
      ),
    },
    {
      key: "price",
      title: "Price",
      customNode: (_, { price }) => <CustomNumberFormat value={price} />,
    },
    {
      key: "isAvailable",
      title: "Is Available",
      customNode: (_, { quantity }) => (
        <Text className="text-xs dark:text-lighter font-semibold">
          {quantity && quantity > 0 ? "Yes" : "No"}
        </Text>
      ),
    },
  ];

  return (
    <DataGrid
      data={products}
      columns={columns}
      className="!shadow-none"
      hasPagination={false}
    />
  );
};

export default OtherProducts;
