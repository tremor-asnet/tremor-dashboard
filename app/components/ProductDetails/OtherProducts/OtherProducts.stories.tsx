import type { Meta, StoryObj } from "@storybook/react";

// Components
import OtherProducts from "./OtherProducts";

// Mocks
import { MOCK_PRODUCTS } from "@/mocks";

const meta = {
  title: "Components/OtherProducts",
  component: OtherProducts,
  tags: ["autodocs"],
} as Meta<typeof OtherProducts>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => (
    <OtherProducts
      products={MOCK_PRODUCTS}
      isAvailable="isAvailable"
      keyword="productName"
    />
  ),
};
