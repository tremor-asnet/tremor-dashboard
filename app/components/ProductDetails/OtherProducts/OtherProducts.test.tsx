// Components
import { render } from "@testing-library/react";
import OtherProducts from "./OtherProducts";

// Mocks
import { MOCK_PRODUCTS } from "@/mocks";

describe("Other Products Table Testing", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <OtherProducts
        products={MOCK_PRODUCTS}
        isAvailable="isAvailable"
        keyword="productName"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
