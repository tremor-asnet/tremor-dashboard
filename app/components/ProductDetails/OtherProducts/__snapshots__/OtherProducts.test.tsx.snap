// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Products Table Testing should match snapshot 1`] = `
<div>
  <div
    class="flex justify-center items-center"
    role="status"
  >
    <svg
      aria-hidden="true"
      class="w-8 h-8 text-transparent animate-[spin_0.4s_linear_infinite] dark:text-gray-600 fill-river-bed-500 dark:fill-white"
      fill="none"
      viewBox="0 0 22 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 3C11 2.44772 11.4477 2 12 2C17.5228 2 22 6.47715 22 12C22 14.7175 20.9147 17.1835 19.1562 18.9849C18.7704 19.3801 18.1373 19.3878 17.7421 19.002C17.3469 18.6162 17.3392 17.983 17.725 17.5879C19.1336 16.1449 20 14.1746 20 12C20 7.58172 16.4183 4 12 4C11.4477 4 11 3.55228 11 3Z"
        fill="currentFill"
      />
    </svg>
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
